ts_src/psbt.ts

```diff
@@ -1746,49 +1736,108 @@ function randomBytes(options?: RngOpts): Buffer {
-  // check if confidential
-  if (blindPrivKey) {
-    return unblindWitnessUtxo(prevout, blindPrivKey);
-  }
-
-  const unblindedInputBlindingData: BlindingData = {
-    value: confidential.confidentialValueToSatoshi(prevout.value).toString(10),
-    ag: prevout.asset.slice(1),
-    abf: ZERO,
-    vbf: ZERO,
+/**
+ * Compute outputs blinders
+ * @param inputsBlindingData the transaction inputs blinding data
+ * @param outputsData data = [satoshis, asset] of output to blind
+ * @returns an array of BlindingData[] corresponding of blinders to blind outputs specified in outputsData
+ */
+export function computeOutputsBlindingData(
+  inputsBlindingData: confidential.BlindingData[],
+  outputsData: Array<[number, string]>,
+): confidential.BlindingData[] {
+  const outputsBlindingData: confidential.BlindingData[] = [];
+  // tslint:disable-next-line:no-shadowed-variable
+  outputsData.slice(0, outputsData.length - 1).forEach(([satoshis, asset]) => {
+    const blindingData: confidential.BlindingData = {
+      satoshis,
+      asset,
+      assetBlinder: randomBlinder(),
+      amountBlinder: randomBlinder(),
+    };
+    outputsBlindingData.push(blindingData);
+  });
+
+  // tslint:disable-next-line:no-shadowed-variable
+  const [satoshis, asset] = outputsData[outputsData.length - 1];
+  const finalBlindingData: confidential.BlindingData = {
+    satoshis,
+    asset,
+    assetBlinder: randomBlinder(),
+    amountBlinder: '',
   };
 
-  return unblindedInputBlindingData;
+  const inputsValues = inputsBlindingData.map(({ satoshis: amount }) =>
+    amount.toString(10),
+  );
+  const outputsValues = outputsData
+    .map(([amount]) => amount.toString(10))
+    .concat(satoshis.toString(10));
+  const inputsAssetBlinders = inputsBlindingData.map(({ assetBlinder }) =>
+    Buffer.from(assetBlinder, 'hex'),
+  );
+  const outputsAssetBlinders = outputsBlindingData
+    .map(({ assetBlinder }) => Buffer.from(assetBlinder, 'hex'))
+    .concat(Buffer.from(finalBlindingData.assetBlinder, 'hex'));
+  const inputsAmountBlinders = inputsBlindingData.map(({ amountBlinder }) =>
+    Buffer.from(amountBlinder, 'hex'),
+  );
+  const outputsAmountBlinders = outputsBlindingData.map(({ amountBlinder }) =>
+    Buffer.from(amountBlinder, 'hex'),
+  );
+
+  const finalAmountBlinder = confidential
+    .valueBlindingFactor(
+      inputsValues,
+      outputsValues,
+      inputsAssetBlinders,
+      outputsAssetBlinders,
+      inputsAmountBlinders,
+      outputsAmountBlinders,
+    )
+    .toString('hex');
+
+  finalBlindingData.amountBlinder = finalAmountBlinder;
+  outputsBlindingData.push(finalBlindingData);
+
+  return outputsBlindingData;
 }
 
-function unblindWitnessUtxo(
-  prevout: WitnessUtxo,
-  blindingPrivKey: Buffer,
-): BlindingData {
-  const unblindProof = confidential.unblindOutput(
-    prevout.nonce,
-    blindingPrivKey,
-    prevout.rangeProof!,
-    prevout.value,
-    prevout.asset,
-    prevout.script,
-  );
+function toBlindingData(
```

Comment:
If we do not export this, is `computeOutputsBlindingData` still usable from outside?

Reply:
yes `computeOutputsBlindingData` uses only `BlindingData`. However, I agree that `toBlindingData` could be useful if you want to use `computeOutputsBlindingData` using private blinding keys.